zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -309,16 +312,24 @@ protected void sendAndReceive(
         // ZAP: Runs the "beforeScan" methods of any ScannerHooks
         parent.performScannerHookBeforeScan(message, this);
 
-        if (isFollowRedirect) {
-            parent.getHttpSender().sendAndReceive(message, getParent().getRedirectRequestConfig());
-        } else {
-            parent.getHttpSender().sendAndReceive(message, false);
+        try {
+            if (isFollowRedirect) {
+                parent.getHttpSender()
+                        .sendAndReceive(message, getParent().getRedirectRequestConfig());
+            } else {
+                parent.getHttpSender().sendAndReceive(message, false);
+            }
+        } catch (IOException e) {
+            // TODO: How to update HistoryRef with error Response?
```

Comment:
@thc202 can you give me any hint here?
How can I update the HistoryRef with error Response? In case HttpMessage has already a HistoryRef and is attached to the Alert

Reply:
Discussed in IRC: No workarounds here. Problem is releated to oast
I guess the way how histRef ist set to HttpMessage is like the following:
.alertFound(alert, null) is triggered by oast here before exception is catched:

https://github.com/zaproxy/zap-extensions/blob/7016bdf3607f31f42ceb3206c86b437131b0452b/addOns/oast/src/main/java/org/zaproxy/addon/oast/ExtensionOast.java#L268

And then a HistRef is created
https://github.com/zaproxy/zaproxy/blob/f68b04e580fac25e2bacecec4bf7365cdc03e5fc/zap/src/main/java/org/zaproxy/zap/extension/alert/ExtensionAlert.java#L189

Which sets automatically the histref member on the Httpmessage

Resulting in that the Histref is set to the message not including the error message which is later set

> we should change oast to allow to manipulate the alert/message before caching it